core/src/main/java/jenkins/slaves/JnlpAgentReceiver.java

```diff
@@ -55,7 +57,16 @@
      * @throws Exception
      *      Any exception thrown from this method will fatally terminate the connection.
      */
-    public abstract boolean handle(String name, JnlpSlaveHandshake handshake) throws IOException, InterruptedException;
+    public abstract boolean handle(String name, JnlpServerHandshake handshake) throws IOException, InterruptedException;
```

Comment:
It still breaks the binary compatibility, because external classes won't implement the new abstract method. Maybe NIT in the case of remoting